### PR TITLE
Add xunit data attribute for finding types that need to be implemented

### PR DESCRIPTION
**Pull request by bash**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -46,3 +46,9 @@ Initial release
 
 ## 0.6.1
 Rerelease of 0.6.0 with fixed nuget package.
+
+## 0.6.2
```
Comment (jnferner) [inline changelog.md]: Make this a breaking change

File: changelog.md [hunk 2]
```diff
@@ -46,3 +46,9 @@ Initial release
...
+- Add `TypesThatNeedToBeImplementedInAssemblyData` attribute which provides a list of types
+  that need to implemented in an assembly to an xUnit theory.
+- Mark `ModuleInterfaceEnumerable`, `ObjectArrayEnumerable`, and `ContainerInterfaceRetriever` as obsolete.
```
Comment (jnferner) [inline changelog.md]: Should read "Remove"

File: Test.Utility.Test/ContainerInterfaceRetrieverTest.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿using System;
+﻿#pragma warning disable 618
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility.Test/ContainerInterfaceRetrieverTest.cs]: This class has been removed.

File: Test.Utility/ModuleInterfaceEnumerable.cs [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
+            return typeof(T)
+                .Assembly
+                .GetTypes()
+                .Where(type => type.IsInterface)
+                .GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility/ModuleInterfaceEnumerable.cs]: Class removed

File: Test.Utility/ModuleInterfaceEnumerable.cs [hunk 5]
```diff
@@ -1,26 +1,27 @@
...
-        {
-            return GetEnumerator();
-        }
-    }
-}
+        where T : IModule
+    {
+        public IEnumerator<Type> GetEnumerator()
+        {
+            return typeof(T)
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility/ModuleInterfaceEnumerable.cs]: Class removed

File: Test.Utility/ObjectArrayEnumerable.cs [hunk 6]
```diff
@@ -1,22 +1,24 @@
...
-        IEnumerator IEnumerable.GetEnumerator()
-        {
-            return GetEnumerator();
-        }
-    }
+        where T : IEnumerable<object>, new()
+    {
+        public IEnumerator<object[]> GetEnumerator()
+        {
+            return new T()
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility/ObjectArrayEnumerable.cs]: Class removed

File: Test.Utility/ObjectArrayEnumerable.cs [hunk 7]
```diff
@@ -1,22 +1,24 @@
...
+                .Select(@interface => new[] { @interface })
+                .GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility/ObjectArrayEnumerable.cs]: Class removed

File: changelog.md (hunk 1 above)
Comment (bash) [reply changelog.md]: Already done

File: changelog.md (hunk 2 above)
Comment (bash) [reply changelog.md]: Already done

File: Test.Utility/Test.Utility.csproj [hunk 8]
```diff
@@ -10,7 +10,7 @@
     </PropertyGroup>
     <PropertyGroup>
         <PackageId>Messerli.Test.Utility</PackageId>
-        <Version>0.6.1</Version>
+        <Version>0.6.2</Version>
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility/Test.Utility.csproj]: Already done

File: Test.Utility.Test/TypesThatNeedToBeImplementedInAssemblyRetrieverTest.cs [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace Messerli.Test.Utility.Test
+{
+    public class TypesThatNeedToBeImplementedInAssemblyRetrieverTest
+    {
+        private const string TestAssemblyName = "TestAssembly";
+
+        private const string EmptyAssemblyName = "EmptyAssembly";
+
+        private static readonly Type[] ImplementableTypesInTestAssembly =
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility.Test/TypesThatNeedToBeImplementedInAssemblyRetrieverTest.cs]: 20e5012

File: Test.Utility.Test/TypesThatNeedToBeImplementedInAssemblyRetrieverTest.cs [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+        private readonly ITestOutputHelper _testOutputHelper;
+
+        public TypesThatNeedToBeImplementedInAssemblyRetrieverTest(ITestOutputHelper testOutputHelper)
+        {
+            _testOutputHelper = testOutputHelper;
+        }
+
+        [Fact]
+        public void FindsOnlyImplementableTypes()
```
[the remark this answers is not in the document]
Comment (bash) [reply Test.Utility.Test/TypesThatNeedToBeImplementedInAssemblyRetrieverTest.cs]: 20e5012

**Pull request merged**